source/guides/recipes/sharing-dependencies.md

```diff
@@ -1,209 +1,112 @@
-
-Next create a `pyproject.toml` file with the following contents:
-
-```toml
-[build-system]
-requires = ["setuptools", "setuptools-scm"]
-build-backend = "setuptools.build_meta"
-
-[project]
-name = "shared_project"
-version = "0.0.1"
-
-[project.scripts]
-app = "app:main"
-```
-
-This file tells Python how to build the project and what will execute when you run the executable called `app`.
-
-For the Nix part of the project you'll create two files: `package.nix` and `default.nix`.
-The actual build recipe will be in `package.nix` and `default.nix` will import this file to perform the build.
-
-First create a `package.nix` file like this:
+Use the `inputsFrom` attribute to `pkgs.mkShell`:
 
 ```nix
+# default.nix
+let
+  pkgs = import <nixpkgs> {};
+  build = pkgs.callpackage ./build.nix {};
+in
 {
-  buildPythonApplication,
-  setuptools-scm,
-  flask,
-}:
-
-buildPythonApplication {
-  pname = "shared_project";
-  version = "0.0.1";
-  format = "pyproject";
-  src = builtins.path { path = ./.; name = "shared_project_source"; };
-  propagatedBuildInputs = [
-    setuptools-scm
-    flask
-  ];
+  inherit build;
+  shell = pkgs.mkShell {
+    inputsfrom = [ build ];
+  };
 }
 ```
 
-The Nix expression in this file is a _function_ that produces a derivation.
-This method of defining builds is a common design pattern in the Nix community, and is the format used throughout the `nixpkgs` repository.
-This particular derivation builds your Python application and ensures that `flask`, the library used to create the web application, is available at runtime.
-
-Note that on line 11 of the `package.nix` file the `src` attribute is set using `builtins.path`.
-This creates a [reproducible source path], and is a good habit to form.
-
-[reproducible source path]: https://nix.dev/recipes/best-practices#reproducible-source-paths
-
-Finally, create a `default.nix` that looks like this:
+Import the `shell` attribute in `shell.nix`:
 
 ```nix
-let
-  pkgs = import <nixpkgs> {};
-in
-  {
-    build = pkgs.python3Packages.callPackage ./package.nix {};
-  }
+# shell.nix
+(import ./.).shell
 ```
 
-The `python3Packages.callPackage` function determines which arguments the function in `package.nix` takes (in this case, `buildPythonApplication`, `setuptools-scm`, and `flask`) then calls the function in `package.nix` with the corresponding attributes from `python3Packages`.
-You can read more about the `callPackage` pattern in the [Nix Pills][nix_pills_callpackage].
-
-Also note that this `default.nix` returns an attribute set with a single attribute called `build`.
-This allows adding more attributes later without breaking existing consumers.
-Try to build this project by running `nix-build -A build`
+## Complete example
 
-[nix_pills_callpackage]: https://nixos.org/guides/nix-pills/callpackage-design-pattern.html
+Assuming your build is defined in `build.nix`:
 
-## Adding development packages
+```nix
+# build.nix
+{ hello }: hello
```

Comment:
This feels a bit too minimal to get the point across. It should at least have some dependencies listed in e.g. `buildInputs` which one could then observe to now also have in the `nix-shell`

Reply:
We could have a nonsensical derivation where the builder returns true, but have hello as a build input... or something like that.

Reply:
How about a `cowsay nix > $out` in a `runCommand`, then you can have cowsay as the dependency :)